PXP-11087: Implement Data Dictionary Table Component

alilahrime opened the pull request:
Jira Ticket: [PXP-11087](https://ctds-planx.atlassian.net/browse/PXP-11087)
implements data dictionary with json data

and [PXP-11261](https://ctds-planx.atlassian.net/browse/PXP-11261)

### New Features
data dictionary table component

### Dependency updates
mantine react table


[PXP-11087]: https://ctds-planx.atlassian.net/browse/PXP-11087?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PXP-11261]: https://ctds-planx.atlassian.net/browse/PXP-11261?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

alilahrime commented:
> See the changes above.

added search for https://ctds-planx.atlassian.net/browse/PXP-11261 and addressed above changes. made some assumptions about how it executes (click search button vs debounce, etc). Let me know if any other parts of the ui/functionality should be modified

craigrbarnes commented:
1. Several linting errors need to be fixed
2. Add padding to the search area like in the wireframe below
3. Search terms should be highlighted. There is a Mantine highlight component that can be used if it is easy.
4. Add borders between each search term like in the wireframe.
5. Clearing search seems to not work

![image](https://github.com/uc-cdis/gen3-frontend-framework/assets/1093780/6dfd66a6-2d3a-42ee-9283-99795b69995f)

alilahrime commented:
> 1. Several linting errors need to be fixed
> 2. Add padding to the search area like in the wireframe below
> 3. Search terms should be highlighted. There is a Mantine highlight component that can be used if it is easy.
> 4. Add borders between each search term like in the wireframe.
> 5. Clearing search appears not to  work
> 
> ![image](https://private-user-images.githubusercontent.com/1093780/325306959-6dfd66a6-2d3a-42ee-9283-99795b69995f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyMDAyNDIsIm5iZiI6MTcxNTE5OTk0MiwicGF0aCI6Ii8xMDkzNzgwLzMyNTMwNjk1OS02ZGZkNjZhNi0yZDNhLTQyZWUtOTI4My05OTc5NWI2OTk5NWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDhUMjAyNTQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjYwM2JjMDFjNDIzMWI0MTBiYjVlZjBkYTFjZDUwOWM4N2Q0YjIzY2RiMmRmYzZmZWJmM2QzMjc5MjQ3OGRlNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.MapJjccHBgsEn6PBRCBHEQvUtZIjZNJqzG8w8zz_t-w)

1. resolved the errors and some of the low hanging warnings
2. added some more padding
3. now using the <Highlight /> component inside autocomplete
4. added borders within the suggestions more similar to the mockup, however theres some exterior space around the whole list that i couldn't completely remove
5. hitting the 'X' currently only clears the search input text, wasnt able to replicate the issue, let me know if i misunderstood or any other requested changes
